client/src/app/backend.service.ts

```diff
@@ -60,55 +58,48 @@ export class BackendService {
   }
 
   startCopy() {
-    let url = `${this.baseEndpoint}startCopy`;
-    let headers = new Headers({}); // request is empty as it relies on data in cookies.
-    let options = new RequestOptions({ headers : headers });
-    return this.http.post(url, '', options)
+    // startCopy needs to be a relative post call for the XSRF token to be included in the header.
+    let url = '/_/startCopy';
+    return this.http.post(url, '')
       .map(res => this.startCopySuccess(res))
       .catch(err => this.handleError(err));
   }
 
-  private listDataTypesSuccess(res: Response) {
-    let body = res.json();
+  private listDataTypesSuccess(res: any) {
     let dataTypes: PortableDataType[] = [];
-    for (var prop in body) {
-      dataTypes.push(new PortableDataType(body[prop], body[prop]));
+    for (var prop in res) {
+      dataTypes.push(new PortableDataType(res[prop], res[prop]));
     }
     return dataTypes;
   }
 
-  private listServicesSuccess(res: Response) {
-    let body = res.json();
-
+  private listServicesSuccess(res: any) {
     let exportServices: ServiceDescription[] = [];
-    let exportData = body['export'];
+    let exportData = res['export'];
     for (var name in exportData) {
       exportServices.push(new ServiceDescription(exportData[name], exportData[name]));
     }
-
     let importServices: ServiceDescription[] = [];
-    let importData = body['import'];
+    let importData = res['import'];
     for (var name in importData) {
       importServices.push(new ServiceDescription(importData[name], importData[name]));
     }
-
     return new ServiceDescriptions(importServices, exportServices);
   }
 
-  private importSetupSuccess(res: Response) {
-    let data = res.json();
+  private importSetupSuccess(res: any) {
```

Comment:
Do we know the object type here or below?

Reply:
It just returns an Object, we can add type checking to the response like so: https://angular.io/guide/http#typechecking-the-response 

But I didn't think to add it to the listDataTypes call since its just a json array, without a tag. 
Added to listServices, importSetup and copySetup